src/main_functions_interactive.c

```diff
@@ -1,185 +1,210 @@
+
+					shell_error("\n\nPlease enter a valid input.\n", color);
+                    log_error("Invalid input.");
+					shell_prompt(false);
 					insertcheck = scanf("%s\n", c);
 				}
 
-				printf("Are you sure you wish to replace \"%s\" with \"%s\"? [y, n] : ", misspelled[i], c);
+				printf("Are you sure you want to replace \"%s\" with \"%s\"? [y, n] : ", misspelled[i], c);
 				userconsent = scanf("%s", sig);
+                log_trace("edit_interactive user entered value: '%s'.", sig);
+
 				if (sig[0] == 'y') {
 					userconsent = 1;
-				} else {
+				}
+
+                else {
 					userconsent = 0;
 				}
 			}
 
-			char* newword = strdup(c);
-			printf("\nReplacing \"%s\" with \"%s\". \n", misspelled[i], newword);
+			char *newword = strdup(c);
+			printf("\nReplacing \"%s\" with \"%s\".\n", misspelled[i], newword);
 			correct_line(line_copy, misspelled[i], newword); //modifies line function
 
          	printf("%s", line_copy);
-        	/* if the line is the last line being edited, the last character is
-        	 * EOF, not \n. */
-         	if (returnflag) printf("\n");
-         	printf("%s", underline_misspelled_sentence(misspelled, line_copy, underline));
 
-        } else if (isdigit(choice[0]) && (atoi(&choice[0]) <= max_no_suggestions)) { // choose suggestion
+        	/*
+             *if the line is the last line being edited,
+             * the last character is EOF, not \n.
+             */
+         	if (returnflag) {
+                printf("\n");
+            }
+
+            if (misspelled[i+1] != NULL)
+                printf("%s", underline_misspelled_sentence(misspelled[i+1], line_copy, underline));
+        } 
+
+        else if (isdigit(choice[0]) && (atoi(&choice[0]) <= max_no_suggestions)) { // choose suggestion
         	// Replace misspelled word with chosen replacement
         	int c = atoi(&choice[0]) - 1;
-        	printf("\nReplacing %s with %s.\n\n", misspelled[i], suggestions[c]);
+        	printf("\nReplacing \"%s\" with \"%s\".\n", misspelled[i], suggestions[c]);
         	correct_line(line_copy, misspelled[i], suggestions[c]); //modifies line function
 
         	/* if the line is the last line being edited, the last character is
         	 * EOF, not \n. */
          	printf("%s", line_copy);
-         	if (returnflag) printf("\n");
-         	printf("%s", underline_misspelled_sentence(misspelled, line_copy, underline));
+         	if (returnflag) {
+                printf("\n");
+            }
+
+            if (misspelled[i+1] != NULL)
+                printf("%s", underline_misspelled_sentence(misspelled[i+1], line_copy, underline));
         }
 
-        i++;	// added loop changer
+        i++; // added loop changer
     }
 
 	return line_copy;
 }
 
 /* See main_functions_interactive.h */
-char** interactive_mode(char* filename, dict_t* dict, bool* quit)
-{
-	char** lines;
+char **interactive_mode(char *filename, dict_t *dict, bool *quit, bool *color) {
```

Comment:
Why is this a `bool *` and not just a `bool`?

Reply:
`*quit` and `*color` are booleans called here because the batch_mode should change its value, initially malloc'd in main.c, in the case that parse_file fails, no file is read and the function returns (this is the conditional checked in line 254). If I just called the value, I wouldn't be able to change it to affect future prints to output. main_page is one of a few functions that can change that value.
I've changed the function calls for functions who won't affect the value, such as the many you commented on for shellstrings, but I think for batch_mnode it might be necessary to do `*quit` and `*color`.